rerun_py/src/python_bridge.rs

```diff
@@ -1372,6 +1373,43 @@ fn new_entity_path(parts: Vec<Bound<'_, pyo3::types::PyString>>) -> PyResult<Str
 
 // --- Properties ---
 
+#[derive(FromPyObject)]
+#[pyo3(from_item_all)]
+struct RecordingProperties {
+    name: Option<String>,
+    started: Option<i64>,
+}
+
+// Set the name of a recording.
+#[pyfunction]
+#[pyo3(signature = (properties, recording=None))]
+fn set_properties(
+    properties: RecordingProperties,
```

Comment:
fine to ship this, but I think I would have just gone with a [`PyAny`](https://pyo3.rs/main/doc/pyo3/types/struct.pyany) and then poke it until it yields the information we want. But that's probably super tedious, don't think I've ever tried.

Reply:
I belief the current approach should offer better IDE support as it is more strictly typed. The downside of course is that we loose some generality. The ideal fix would be to move the archetypes to the `pyo3` boundary to get all the nice things.